lib/convection/model/template.rb

```diff
@@ -76,6 +82,10 @@ def resource(name, &block)
         resources[name] = r
       end
 
+      def resource_collection(name, &block)
+        resource_collections[name] = Model::Template::ResourceCollection.new(name, self, &block)
+      end
```

Comment:
> What use case do you see for this?

@bturner-r7 I'm glad you spotted this and I meant to bring it up as a "concern". I wasn't really sure on whether we should expose `resource_collection` as a DSL method because like you said:

> Why would you want to define resources inside an inline ResourceCollection as opposed to directly in a template?

I haven't been able to answer this question. We can easily take this out and remove the not implemented error since the nested use-case wouldn't need to be prevented.